### PR TITLE
doc: Adds cmake build options for optionally skipping few components

### PR DESCRIPTION
**Pull request by ideepika**
Signed-off-by: Deepika Upadhyay <dupadhya@redhat.com>

## Checklist
- [ ] References tracker ticket
- [x] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

**Comment by ideepika** [comment]
jenkins test docs

File: doc/dev/quick_guide.rst [hunk 1]
```diff
@@ -15,6 +15,15 @@ the result behaves as expected.
 
        $ ./run-make-check.sh
 
+Optionally if you need a specific component of Ceph to work on,
```
Comment (tchaikov) [inline doc/dev/quick_guide.rst]:
```suggestion
Optionally if you want to work on a specific component of Ceph,
```

**Comment by ideepika** [comment]
> Can you squash your commits into one? Otherwise, LGTM.

Yep, sure!

**Comment by ideepika** [comment]
jenkins retest this please

**Comment by neha-ojha** [comment]
jenkins render docs

**Pull request merged**